### DISCUSSION TITLE
Questions about realworld project

### DISCUSSION THREAD

**Discussion by sukam09** [realworld-apps/realworld#964] [Q&A]
Hi, I have several questions about the project.

1. The correct URL of profile page is ambiguous because the examples of frontend spec and demo app are slightly different such as [/#/profile/:username](https://realworld-docs.netlify.app/docs/specs/frontend-specs/routing)(in frontend spec) and [/#/@:username](https://demo.realworld.io/#/@sukam09)(in demo app). So please let me know which is right guideline for our implementations.

2. ~~I don't understand why "token" is in the PUT /user in User and Authentication in demo API. Shouldn't it be changed to "password"? Or is there anything proper use of token field in this endpoint which I don't know? That is, how can I change password with/without PUT /user API?
+I found [this section](https://realworld-docs.netlify.app/docs/specs/backend-specs/endpoints#update-user) a moment ago and also checked password field works so it may be miss of Swagger docs. Please check it.~~
It was duplicated. Here is link: #950

3. ~~In docs, there is expectation about using libraries/frameworks which have more than 300 GitHub stars. ([link](https://realworld-docs.netlify.app/docs/implementation-creation/expectations#other-expectations))
Is this rule applied to all libraries/frameworks in my implementation? Can I use both [recoil](https://github.com/facebookexperimental/Recoil)(17.2k stars) and [recoil-persist](https://github.com/polemius/recoil-persist)(259 stars)?~~
I decided not to use recoil-persist.

4. Should I consider outer intervention of API in this implementation? e.g., updating username with another demo app, and so on.

Thank you.

**Comment by kkminseok** [comment]
I'm implementing a fullstack(spring+ vue)
but, I found out that many requirements were different.

Like In Body

The correct URL of profile page is ambiguous because the examples of frontend spec and demo app are slightly different such as [/#/profile/:username](https://realworld-docs.netlify.app/docs/specs/frontend-specs/routing)(in frontend spec) and [/#/@:username](https://demo.realworld.io/#/@sukam09)(in demo app). So please let me know which is right guideline for our implementations.


and

- in demo app, In setting Page `update User ` API Error Handling is not in Process (e.g Duplicate username, Duplicate Email ...) 
- demo app CSS Different <https://realworld-docs.netlify.app/docs/specs/frontend-specs/templates#home> Template (Home template, Setting Template ... )

**Reply by geromegrignon** [reply]
Hi @kkminseok,

Please refer to the specs only.
I need to update the old demo app but been busy lately.

About CSS, Can you illustrate the difference? It uses the same structure from a quick look.

**Reply by kkminseok** [reply]
@geromegrignon 
Thank you for your quick response.

## in demo app main home

![image](https://user-images.githubusercontent.com/30401054/202413619-b61c4c64-0845-49e5-afcb-dc16721255f8.png)

## in template main home

![image](https://user-images.githubusercontent.com/30401054/202415275-6ad11ac0-b356-4674-9658-1fb8dae96805.png)

------

## in demo app setting page

![image](https://user-images.githubusercontent.com/30401054/202413890-5a31fe46-2b65-4e7d-bd1e-bf4051557d6e.png)

## in template setting page not inclue `logout` Button

![image](https://user-images.githubusercontent.com/30401054/202414103-4ad069fb-995b-4370-a4af-21a664ea04ff.png)

-----

**Reply by geromegrignon** [reply]
Thanks, i see., I'll update the templates.

**Reply by sukam09** [reply]
@geromegrignon
Thank you for your answer. Because few days passed from that day I uploaded this question, most of my questions were solved.
I tuned my all implementations to frontend specs (not demo app). And also thanks to @kkminseok for reminding this issue again.

Now I almost completed my demo app but one more question left. Should I consider intervention of another demo app? For example, if someone changes his/her user information with the app (not mine), backend data will be changed. Therefore, when he or she logs in with my app, I want to know whether the changed info should be applied or not. So I hope to get your advices in this situation.

Thanks.

**Reply by geromegrignon** [reply]
@sukam09 sorry for being so late on your previous questions. I totally missed it.

If you are using api.realworld.io API, then any changes made on the official demo app or by someone consuming this API will be reflected on your demo app.

**Reply by geromegrignon** [reply]
@kkminseok about the home page, I can't reproduce content being aligned wrongly: here is an example NOT reproducing it by importing the stylesheet and the template code: https://stackblitz.com/edit/web-platform-aipzih?file=index.html

**Reply by kkminseok** [reply]
> @geromegrignon Hi.

I Think during the main.css import process error.

![image](https://user-images.githubusercontent.com/30401054/203207781-25125c21-eac6-4ca4-bb8e-ee51274f3c70.png)

util, css apply success
